core/controllers/email_dashboard_test.py

```diff
@@ -152,6 +184,268 @@ def setUp(self):
         self.set_admins(
             [self.SUBMITTER_USERNAME, self.NEW_SUBMITTER_USERNAME])
 
+    def test_handler_with_invalid_num_queries_to_fetch_raises_error_400(self):
+        self.login(self.SUBMITTER_EMAIL)
+
+        response = self.get_json(
+            '/emaildashboarddatahandler',
+            params={'num_queries_to_fetch': '-5'},
+            expected_status_int=400)
+        self.assertEqual(
+            response['error'], '400 Invalid input for query results.')
+
+        self.logout()
+
+    def test_email_dashboard_data_handler(self):
+        self.login(self.SUBMITTER_EMAIL)
+
+        response = self.get_json(
+            '/emaildashboarddatahandler',
+            params={'num_queries_to_fetch': 1})
+        self.assertEqual(response['recent_queries'], [])
+
+        query_id = user_query_services.save_new_query_model(
+            self.submitter_id, inactive_in_last_n_days=10,
+            created_at_least_n_exps=5,
+            has_not_logged_in_for_n_days=30)
+
+        response = self.get_json(
+            '/emaildashboarddatahandler',
+            params={'num_queries_to_fetch': 1})
+
+        self.assertEqual(response['recent_queries'][0]['id'], query_id)
```

Comment:
Please check length first since we also want to make sure only one is returned:

```python
self.assertEqual(len(response['recent_queries']), 1)
recent_query = response['recent_queries'][0]
self.assertEqual(recent_query['id'], query_id)
self.assertEqual(recent_query['status'], feconf.USER_QUERY_STATUS_PROCESSING)
```

Reply:
Done